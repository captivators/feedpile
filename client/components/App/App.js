import React from 'react';
import ReaderList from '../ReaderList/ReaderList';
import Sidebar from '../Sidebar/Sidebar';
import Navbar from '../Navbar/Navbar';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import injectTapEventPlugin from 'react-tap-event-plugin';
import { getArticlesFromDb } from '../../actions';
import { connect } from 'react-redux'

// Needed for onTouchTap
// http://stackoverflow.com/a/34015469/988941
injectTapEventPlugin();

import './app.css';

class App extends React.Component {
  componentDidMount() {
    this.props.getArticlesFromDb();
  }

  render() {
    const { history } = this.props;
    return (
      <MuiThemeProvider>
          <div className='app-container'>
            <Navbar />
            <Sidebar />
            <ReaderList history={history}/>
          </div>
      </MuiThemeProvider>
    );
  }
}
export const Unwrapped = App;
export default connect(null, {getArticlesFromDb})(App);
