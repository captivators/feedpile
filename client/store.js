import {createStore, compose, applyMiddleware} from 'redux';
import rootReducer from './root-reducer';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './root-saga';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, compose (
  applyMiddleware(logger, sagaMiddleware)
));

sagaMiddleware.run(rootSaga);
export default store;
