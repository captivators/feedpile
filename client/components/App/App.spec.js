import React from 'react';
import {Provider} from 'react-redux';
import store from '../../store';
import { Unwrapped as UnwrappedApp } from './App'
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

test('App take a snapshot', () => {
  const component = shallow(<UnwrappedApp />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('App should render a Sidebar and ReaderList', () => {
  const component = render(<Provider store={store}><UnwrappedApp history={{}}/></Provider>);
  expect(component.find('.sidebar-container').length).toEqual(1);
  expect(component.find('.reader-list-container').length).toEqual(1);
});

