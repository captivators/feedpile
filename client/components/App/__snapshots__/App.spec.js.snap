// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App take a snapshot 1`] = `
<MuiThemeProvider>
  <div
    className="app-container"
  >
    <Navbar />
    <Connect(Sidebar) />
    <Connect(ReaderList) />
  </div>
</MuiThemeProvider>
`;
